feat: canonicalize BitVec to arithmetic before running ring_nf

AtticusKuhn opened the pull request:
There are 2 main kinds of operations on BitVecs

1. Boolean operations (^^^, &&&, |||) which are solved by extensionality
2. Arithmetic operations (+, -) which are solved by `ring_nf`

The purpose of this commit is to convert boolean operations to arithmetic operations and then
solve the arithmetic with the `ring_nf` tactic. See `bitvec_AddSub_1202` in `SSA/Projects/InstCombine/AliveStatements.lean` for an example of this.

`bitvec_AddSub_1202` was not solved by the `alive_auto` tactic before this commit, and now it is solved by the `alive_auto` after this commit.


I added the line

```lean
 try solve | (simp only   [← BitVec.allOnes_sub_eq_xor,  BitVec.negOne_eq_allOnes'];  ring_nf)
```

to the `alive_auto` tactic.

Commit:
Re-add comment

tobiasgrosser commented on SSA/Projects/InstCombine/TacticAuto.lean:
```diff
@@ -91,6 +91,14 @@ macro "simp_alive_bitvec": tactic =>
+        2. Arithmetic operations (+, -)
+        The purpose of the below line is to convert boolean operations to arithmetic operations and then
+        solve the arithmetic with the `ring_nf` tactic. See `bitvec_AddSub_1202` for an example of this
+        -/
```
```suggestion
        The purpose of the below line is to convert boolean operations to arithmetic operations and then
        solve the arithmetic with the `ring_nf` tactic.
        -/
```

If you would like to have an example, I would suggest to inline it directly.

tobiasgrosser replied:
Also, the lines seem to be a bit long. Are they shorter than 100 characters?

tobiasgrosser commented:
Overall, this patch already looks really good. I just have some minor comments.

>  try solve | (simp only   [← BitVec.allOnes_sub_eq_xor,  BitVec.negOne_eq_allOnes'];  ring_nf)

This line has unnecessary whitespace.

tobiasgrosser commented on SSA/Projects/InstCombine/TacticAuto.lean:
```diff
@@ -91,6 +91,14 @@ macro "simp_alive_bitvec": tactic =>
+        The purpose of the below line is to convert boolean operations to arithmetic operations and then
+        solve the arithmetic with the `ring_nf` tactic. See `bitvec_AddSub_1202` for an example of this
+        -/
+        try solve | (simp only   [← BitVec.allOnes_sub_eq_xor,  BitVec.negOne_eq_allOnes'];  ring_nf)
```
```suggestion
        try solve | (simp only [← BitVec.allOnes_sub_eq_xor, BitVec.negOne_eq_allOnes']; ring_nf)
```

Commit:
Make Lines Shorter, Remove Whitespace

Improve the PR in accordance with Tobias' changes.
1. Make the lines fit on 100 columns.
2. Remove unnessecary whitespace.

AtticusKuhn commented:
Sounds good. I fixed the PR in accordance with your feedback. Is this PR ready to be merged?

tobiasgrosser commented:
Nice. Only 35 failing. Before we had 38 failing.

AnotherAlexHere commented:
Hello, I have been asked to inform you that your last commit should be in a different pull request.
